src/routes/customer-resource/customer-resource-show.js

```diff
@@ -1,53 +1,160 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import fetch from 'isomorphic-fetch';
+import { connect } from 'react-redux';
+import { handleActions } from 'redux-actions';
+import  { Observable }  from 'rxjs/Observable';
+import { bindActionCreators } from 'redux'
+
+import 'rxjs/add/observable/from';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/switchMap';
 
 import View from '../../components/customer-resource-show';
 
-export default class CustomerResourceShowRoute extends Component {
+const ActionTypes = {
+  'CUSTOMER_RESOURCE_SHOW_ERROR': 'CUSTOMER_RESOURCE_SHOW_ERROR',
+  'CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_REJECT': 'CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_REJECT',
+  'CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_RESOLVE': 'CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_RESOLVE',
+  'CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED': 'CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED',
+  'CUSTOMER_RESOURCE_SHOW_LOADED': 'CUSTOMER_RESOURCE_SHOW_LOADED',
+  'CUSTOMER_RESOURCE_SHOW_LOAD': 'CUSTOMER_RESOURCE_SHOW_LOAD'
+};
+
+class CustomerResourceShowRoute extends Component {
   static propTypes = {
     match: PropTypes.shape({
       params: PropTypes.shape({
         packageId: PropTypes.string.isRequired,
         titleId: PropTypes.string.isRequired,
         vendorId: PropTypes.string.isRequired
       }).isRequired
-    }).isRequired,
-    resources: PropTypes.shape({
-      showCustomerResource: PropTypes.shape({
-        records: PropTypes.arrayOf(PropTypes.object)
-      })
-    })
+    }).isRequired
   };
 
-  static manifest = Object.freeze({
-    showCustomerResource: {
-      type: 'okapi',
-      path: 'eholdings/vendors/:{vendorId}/packages/:{packageId}/titles/:{titleId}',
-      pk: 'titleId'
-    }
-  });
+  static contextTypes = {
+    addReducer: PropTypes.func.isRequired,
+    addEpic: PropTypes.func.isRequired
+  }
+
+
+  componentWillMount() {
+    let { okapi } = this.props;
+
+    let headers = {
+      'Content-Type': 'application/json',
+      'X-Okapi-Tenant': okapi.tenant
+    };
+
+
+    const ToggleSelectionEpic = (action$, { getState }) =>
+          action$.ofType(ActionTypes.CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED)
+          .mergeMap(action => {
+            let state = getState().customerResourceShow;
+            let body = JSON.stringify({isSelected: state.isSelected});
+            return Observable.from(fetch(state.endpoint, { headers, method: 'PUT', body }));
+          })
+          .map(json => ({ type: ActionTypes.CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_RESOLVE }))
+          .catch(error => ({type: ActionTypes.CUSTOMER_RESOURCE_SHOW_TOGGLE_SELECTED_REJECT }));
```

Comment:
This will actually terminate the action stream when it's set on the `action$` itself. [Source](https://redux-observable.js.org/docs/recipes/ErrorHandling.html)

Reply:
You'll want this chained from `Observable.from(...)` instead

Reply:
Is it not chained from `Observable.from` ? 

```js
Observable.from(x).map(y).catch(z);
```

Reply:
This is currently chained from the `action$` stream that results in `Observable.from`. 

From the linked docs: "Here we placed the `.catch()` inside our `.mergeMap()`, but after our AJAX call; this is important because if we let the error reach the `action$.ofType()`, it will terminate it and no longer listen for new actions."

Reply:
since the result of the `mergeMap` is returning an Observable, the way I understood it is that `map` and `catch` were both hanging off that Observable.  in the example they're being called inside the mergeMap and that's not the case here, right?

Reply:
Right, they should be called inside of the merge map so the error never reaches the `action$` stream. While the observable is returned inside the stream, it doesn't create a new stream. `mergeMap` returns the current stream with the return value as it's `subject`, in this case that's the `action$.ofType` stream with the observable as it's `subject`.

What we want to return inside of `mergeMap` is an observable that handles it's own stream errors

Reply:
I hope that made sense

Reply:
Oops, I read it wrong. Yes, the map and catch are hanging off of the `switchMap`, for some reason I could swear they were off the the ajax observable.